docs/docs/dev_docs/contracts/syntax/slow_updates_tree.md

```diff
@@ -0,0 +1,219 @@
+---
+title: Slow Updates Tree
+---
+
+Slow Updates Tree is a data structure that allows for public data to be accessed in both private and public domains. Read the high level overview in the [concepts section](../../../concepts/foundation/communication/public_private_calls/slow_updates_tree.md).
+
+The Slow Updates Tree works by having a current tree and a pending tree, and replacing the current tree with the pending tree after an epoch has passed. Public functions can read directly from the current tree, and private functions can perform a membership proof that values are part of a commitment to the current state of the tree.
+
+On this page you will learn:
+
+1. About the SlowTree.nr smart contract
+2. The components involved in using the Slow Updates Tree 
+3. How you can integrate it into your own smart contract
+
+Scroll down to view [integration guide](./slow_updates_tree.md#how-to-integrate-a-slow-updates-tree) and [reference](./slow_updates_tree.md#reference).
+
+# Exploring an example integration through a **`TokenBlacklist`** Smart Contract
+
+The `TokenBlacklist` contract is a token contract that does not allow blacklisted accounts to perform mints or transfers. To achieve this, it interacts with a Slow Updates Tree primarily through the `SlowMap` interface. There are four main components involved in this smart contract:
+
+1. **TokenBlacklist.nr Contract:** This is the primary smart contract that utilizes the Slow Updates Tree for managing blacklisted addresses
+2. **SlowMap Interface**: This interface is used within the `TokenBlacklist` contract to interact with the Slow Updates Tree. It provides methods for reading and updating values in the tree in both public and private contexts.
+3. **SlowTree.nr Contract**: This is a smart contract that instantiates a slow updates tree and allows us to access and manipulate its contents.
+4. **SlowMap type**: This is a type in the Azetc library that is utilized by the SlowTree contract.
+
+## TokenBlacklist Contract
+
+You can find the full code for the TokenBlacklist smart contract [here](https://github.com/AztecProtocol/aztec-packages/tree/master/yarn-project/noir-contracts/src/contracts/token_blacklist_contract).
+
+### Importing SlowMap
+
+The contract first imports the **`SlowMap`** interface:
+
+#include_code interface yarn-project/noir-contracts/src/contracts/token_blacklist_contract/src/main.nr rust
+
+The **`SlowMap`** interface allows the contract to interact with its attached SlowTree. It abstracts these functions so they do not have to be implemented in the TokenBlacklist contract.
+
+### Constructor and Initialization of the Slow Updates Tree
```

Comment:
ah but you are not initializing the SU tree - you are initializing the token contract to tell which tree address to talk to. This heading is misleading and makes me think I have to build the tree

Reply:
So in theory there can be many versions of the slow updates tree?

Reply:
Could epoch length be configurable?

Reply:
Could epoch length be configurable?